lib/langchain/llm/openai.rb

```diff
@@ -119,24 +129,38 @@ def compose_parameters(model, params)
     def compose_chat_messages(prompt:, messages:, context:, examples:)
       history = []
 
-      history.concat transform_messages(examples) unless examples.empty?
+      add_examples_to_history(examples, history)
```

Comment:
I'm not convinced slicing this method in these smaller `add_..._to_history` methods makes it more readable/maintainable. First of all, it is a **conditional** add whatever to history, and the method name does not hit there's a condition. So I would move these checks `unless examples.empty?` back to `compose_chat_messages` method.
And if you move those back the only "logic" that goes to those simple add methods is just `history.concat(...)` which is already pretty self-explanatory and there's not a ton of sense to create these tiny methods.

Reply:
- I agree the `history.concat(...)` are self-explanatory but there are many of them, and they add noises to this private interface, the output of this interface is to compose the chat messages. The noise details should be just extracted and put in separate methods, later, if we want to expose this private interface outside, it will make it easier to maintain. 😄